akka-actor/src/main/scala/akka/actor/dungeon/FaultHandling.scala

```diff
@@ -70,7 +70,7 @@ private[akka] trait FaultHandling { this: ActorCell =>
           val ex = PreRestartException(self, e, cause, optionalMessage)
           publish(Error(ex, self.path.toString, clazz(failedActor), e.getMessage))
         } finally {
-          clearActorFields(failedActor, recreate = true)
+          clearActorFields(failedActor, clearContext = true, recreate = true)
```

Comment:
I don't think this is good. We should never clear the context, because with this it looks like there is a short time window where the context is null during restarts.

The reason this is not easy to change is that there are some checks in FaultHandling.scala `context ne null` and `context == null`

This was also discussed in https://github.com/akka/akka/pull/26745#discussion_r275652391

Reply:
So, we would have to find out how to change `FaultHandling` to not require that check as an alternative?